### ISSUE TITLE
After blacklisting "ideapad_laptop" wifi don't detect any network

### ISSUE THREAD

**Issue by SkopeTN** [kfechter/LegionY530Ubuntu#15]
After blacklisting "ideapad_laptop" wifi is enabled, but can't detect any network.

**Comment by kfechter** [comment]
Strange, Mine finds the networks nearly instantly after blacklisting. Are you using 18.04.1 or 18.10?

**Comment by kfechter** [comment]
Also, trying to figure out if the patch for ideapad_laptop is in Ubuntu's kernel yet there was a patch made that makes it so blacklisting isn't required.

**Comment by kfechter** [comment]
It is not in 18.x or 19.x daily builds yet. might be possible to install an updated kernel.

**Comment by SkopeTN** [comment]
Thank you for your response ! i'm currently using the 18.04 LTS, same result when i blacklist it or when i simply deactivate it with "sudo rmmod ideapad_laptop". Network still fetching endlessly.

**Comment by kfechter** [comment]
hmmmm, strange. I was unable to reproduce this behavior on 18.04 or 18.10. 

looks like the idepad_laptop module didn't make the merge window for 4.19 or 4.19.1, but it did make it into 4.20-rc1 so it should be in kernel 4.20

**Comment by popvladaurel** [comment]
My two cents here: got the same issue after blacklisting ideapad_laptop, and managed to solve it by following the answer for [this askubuntu question](https://askubuntu.com/questions/1051381/wifi-realtek-r8822be-not-working-ubuntu-18-04-kernel-4-15-0-20). After making the change, I also had to shutdown the laptop, unplug the charger, leave it alone for a few minutes (5 in my case), plug it back in and, after powering it back on, I was able to see my network and connect to it successfully.

TLDR: `echo "options r8822be aspm=0"  | sudo tee /etc/modprobe.d/r8822be.conf`

**Comment by SkopeTN** [comment]
Thank you very much @popvladaurel !! it finally resolved my problem !

**Comment by kfechter** [comment]
Interesting, do your Legion models not have Intel wifi cards?

**Comment by SkopeTN** [comment]
It's a Realtek for me. I solved the network issue, but still have problem with HDMI that does not detect any monitor if you guys have some ideas for that, i'd really appreciate it !

Thanks again

**Comment by popvladaurel** [comment]
@SkopeTN I was only able to use the HDMI and miniDP outputs by switching to the nVidia display adapter with prime-select nvidia. These ports do not seem to be active when using the intel integrated graphics, xrander lists them only when using nVidia :( 

@kfechter Realtek here aswell, obviously, using the r8822be driver

**Comment by kfechter** [comment]
For the HDMI, it makes sense that it only works with the Nvidia dGPU enabled as all of the external connectors are generally connected to the discrete card in an optimus equipped laptop. 

I only got nvidia output working by using the nvidia-drivers-390 package with prime-select nvidia.

Didn't realize they were shipping Legions with realtek cards. I have the Intel 9560, so that explains why I don't have any issues after blacklist. I can add a section to the guide when I get a chance.

**Comment by kafran** [comment]
@kfechter https://github.com/teejee2008/ukuu

**Comment by kfechter** [comment]
@kafran the next version of this guide will include using ukuu :D 

I am currently running 18.04 with 4.20 manually installed through the  .deb files.

**Comment by kfechter** [comment]
The guide has been updated to include both Updating the kernel with ukuu and the steps to fix realtek wifi cards. Closing this issue as resolved.
